src/optimizer/rules/rules.py

```diff
@@ -0,0 +1,629 @@
+        super().__init__(RuleType.LOGICAL_CREATE_UDF_TO_PHYSICAL, pattern)
+
+    def promise(self):
+        return Promise.LOGICAL_CREATE_UDF_TO_PHYSICAL
+
+    def check(self, before: Operator, context: OptimizerContext):
+        return True
+
+    def apply(self, before: LogicalCreateUDF, context: OptimizerContext):
+        after = CreateUDFPlan(before.name, before.if_not_exists, before.inputs,
+                              before.outputs, before.impl_path, before.udf_type)
+        return after
+
+
+class LogicalInsertToPhysical(Rule):
+    def __init__(self):
+        pattern = Pattern(OperatorType.LOGICALINSERT)
+        # pattern.append_child(Pattern(OperatorType.DUMMY))
+        super().__init__(RuleType.LOGICAL_INSERT_TO_PHYSICAL, pattern)
+
+    def promise(self):
+        return Promise.LOGICAL_INSERT_TO_PHYSICAL
+
+    def check(self, before: Operator, context: OptimizerContext):
+        return True
+
+    def apply(self, before: LogicalInsert, context: OptimizerContext):
+        after = InsertPlan(before.video_catalog_id,
+                           before.column_list, before.value_list)
+        return after
+
+
+class LogicalLoadToPhysical(Rule):
+    def __init__(self):
+        pattern = Pattern(OperatorType.LOGICALLOADDATA)
+        # pattern.append_child(Pattern(OperatorType.DUMMY))
+        super().__init__(RuleType.LOGICAL_LOAD_TO_PHYSICAL, pattern)
+
+    def promise(self):
+        return Promise.LOGICAL_LOAD_TO_PHYSICAL
+
+    def check(self, before: Operator, context: OptimizerContext):
+        return True
+
+    def apply(self, before: LogicalLoadData, context: OptimizerContext):
+        after = LoadDataPlan(before.table_metainfo, before.path)
+        return after
+
+
+class LogicalGetToSeqScan(Rule):
+    def __init__(self):
+        pattern = Pattern(OperatorType.LOGICALGET)
+        # pattern.append_child(Pattern(OperatorType.DUMMY))
+        super().__init__(RuleType.LOGICAL_GET_TO_SEQSCAN, pattern)
+
+    def promise(self):
+        return Promise.LOGICAL_GET_TO_SEQSCAN
+
+    def check(self, before: Operator, context: OptimizerContext):
+        return True
+
+    def apply(self, before: LogicalGet, context: OptimizerContext):
+        after = SeqScanPlan(before.predicate, before.target_list)
+        after.append_child(StoragePlan(before.dataset_metadata))
+        return after
+
+
+class LogicalProjectToSeqScan(Rule):
+    def __init__(self):
+        pattern = Pattern(OperatorType.LOGICALPROJECT)
+        super().__init__(RuleType.LOGICAL_PROJECT_TO_SEQSCAN, pattern)
+
+    def promise(self):
+        return Promise.LOGICAL_PROJECT_TO_SEQSCAN
+
+    def check(self, before: Operator, context: OptimizerContext):
+        return True
+
+    def apply(self, before: LogicalProject, context: OptimizerContext):
+        after = SeqScanPlan(None, before.target_list)
+        return after
+
+
+class LogicalFilterToSeqScan(Rule):
+    def __init__(self):
+        pattern = Pattern(OperatorType.LOGICALFILTER)
```

Comment:
@xzdandy @gaurav274 
Can you check implementation rules here? I am not sure whether the pattern needs a Dummy, and how many? 
Can you also proofread other implementation rules that I added regarding the same issue?

Reply:
I will take a look later today.

Reply:
Where those mentioned integration tests?

Reply:
https://github.com/georgia-tech-db/eva/tree/master/test/integration_tests